pkg/agent/route/route_linux.go

```diff
@@ -2813,122 +2829,292 @@ func (c *Client) DeleteNodeNetworkPolicyIPTables(iptablesChains []string, isIPv6
+	switch ipProtocol {
+	case knftables.IPv4Family:
+		ipAddr = "ipv4_addr"
+		ip = "ip"
+		protocol = "protocol"
+		nodePortDNATVIP = config.VirtualNodePortDNATIPv4.String()
+		svcSNATVIP = config.VirtualServiceIPv4.String()
+		nodePortSet = antreaNFTablesSetNodePort
+		externalIPSet = antreaNFTablesSetExternalIP
+	case knftables.IPv6Family:
+		ipAddr = "ipv6_addr"
+		ip = "ip6"
+		protocol = "nexthdr"
+		nodePortDNATVIP = config.VirtualNodePortDNATIPv6.String()
+		svcSNATVIP = config.VirtualServiceIPv6.String()
+		nodePortSet = antreaNFTablesSetNodePort6
+		externalIPSet = antreaNFTablesSetExternalIP6
+	}
+
+	tx.Add(&knftables.Set{
+		Name:    nodePortSet,
+		Type:    ipAddr + " . inet_proto . inet_service",
+		Comment: ptr.To("Set containing NodePort tuples (ip, protocol, port)"),
+	})
+	tx.Add(&knftables.Set{
+		Name:    externalIPSet,
+		Type:    ipAddr,
+		Comment: ptr.To("Set containing external IPs"),
+	})
+	for _, set := range []string{nodePortSet, externalIPSet} {
+		// Flush the sets if it exists.
+		tx.Flush(&knftables.Set{
+			Name: set,
+		})
+		// Restore the elements.
+		c.serviceNFTablesSets[set].Range(func(_, value any) bool {
+			elements := value.([]*knftables.Element)
+			for _, element := range elements {
+				tx.Add(element)
+			}
+			return true
+		})
+	}
+
+	tx.Add(&knftables.Rule{
+		Chain: antreaNFTablesRawChainPreroutingProxyAll,
+		Rule: knftables.Concat(
+			ip, "daddr", "@", externalIPSet,
+			"counter",
+			"notrack",
+		),
+		Comment: ptr.To("Do not track request packets destined to external IPs"),
+	})
+	tx.Add(&knftables.Rule{
+		Chain: antreaNFTablesRawChainPreroutingProxyAll,
+		Rule: knftables.Concat(
+			ip, "saddr", "@", externalIPSet,
+			"counter",
+			"notrack",
+		),
+		Comment: ptr.To("Do not track reply packets sourced from external IPs"),
+	})
+	tx.Add(&knftables.Rule{
+		Chain: antreaNFTablesRawChainOutputProxyAll,
+		Rule: knftables.Concat(
+			ip, "daddr", "@", externalIPSet,
+			"counter",
+			"notrack",
+		),
+		Comment: ptr.To("Do not track request packets destined to external IPs"),
+	})
+
+	tx.Add(&knftables.Rule{
+		Chain: antreaNFTablesNatChainPreroutingProxyAll,
+		Rule: knftables.Concat(
+			ip, "daddr", ".", ip, protocol, ".", "th dport", "@", nodePortSet,
+			"counter",
+			"dnat", "to", nodePortDNATVIP,
+		),
+		Comment: ptr.To("DNAT external to NodePort packets"),
+	})
+	tx.Add(&knftables.Rule{
+		Chain: antreaNFTablesNatChainOutputProxyAll,
+		Rule: knftables.Concat(
+			ip, "daddr", ".", ip, protocol, ".", "th dport", "@", nodePortSet,
+			"counter",
+			"dnat", "to", nodePortDNATVIP,
+		),
+		Comment: ptr.To("DNAT local to NodePort packets"),
+	})
+	tx.Add(&knftables.Rule{
+		Chain: antreaNFTablesNatChainPostroutingProxyAll,
+		Rule: knftables.Concat(
+			ip, "saddr", svcSNATVIP,
+			"counter",
+			"masquerade",
+		),
+		Comment: ptr.To("DNAT local to NodePort packets"),
+	})
+}
+
+func (c *Client) nftablesTrafficAcceleration(tx *knftables.Transaction, ipProtocol knftables.Family) {
```

Comment:
just to confirm: the code in this function is just extracted and not changed, right?